libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Utilities/CSFInternalDefines.m

```diff
@@ -76,12 +76,8 @@ CSFParameterStyle CSFRequiredParameterStyleForHTTPMethod(NSString *method) {
 Class CSFClassFromEncoding(NSString *encoding) {
     Class result = nil;
     if ([encoding hasPrefix:@"@"]) {
-        if (encoding.length > 2) {
-            NSString *className = [[encoding substringFromIndex:2] stringByReplacingOccurrencesOfString:@"\"" withString:@""];
-            result = NSClassFromString(className);
-        } else {
-            result = [NSObject class];
-        }
+        NSString *className = [[encoding substringFromIndex:2] stringByReplacingOccurrencesOfString:@"\"" withString:@""];
```

Comment:
If we were losing something that had been there for a while I would agree. I added the length check at Michaels recommendation with my previous PR as a potential solution to the issue I reported. Since it did not do anything for that problem, I would prefer to remove it and Michael can add it himself if he feels it is required for some other problem.